### DISCUSSION TITLE
Raise event on cancel dagster run

### DISCUSSION THREAD

**Discussion by programmervarunshridhar** [dagster-io/dagster#7852] [Q&A]
Hello all,

I have a situation where in, on a dagster run cancellation, I have to raise an event (DagsterRunCancelledEvent) and listeners who are listening to this event will go ahead and do the clean up.

Is there a way to pull this off in dagster?

Appreciate your time and help.

**Comment by yuhan** [comment]
Hi! Could you provide more context on your use case? (e.g. code snippet for how you'd raise the event, how you expect the listeners to interact with Dagster?)

Assuming you're looking for a way to act on run cancelled events, one way to do so in Dagster context is to set up a [run status sensor](https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors) which listens to Dagster's run event storage. For example, you can set up a `my_slack_on_run_canceled` sensor as:
```
from dagster import run_status_sensor, RunStatusSensorContext, DagsterRunStatus

@run_status_sensor(pipeline_run_status=DagsterRunStatus.CANCELED)
def my_slack_on_run_canceled(context: RunStatusSensorContext):
    slack_client = WebClient(token=os.environ["YOUR_TOKEN"])
    slack_client.chat_postMessage(
        channel="#alert-channel",
        message=f'Job "{context.pipeline_run.pipeline_name}" was cancelled.',
    )
```
